server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -225,17 +247,33 @@ ActionListener<SearchResponse> transformResponseListener(
             final SearchResponseProcessor processor = searchResponseProcessors.get(i);
 
             responseListener = ActionListener.wrap(r -> {
+                ProcessorExecutionDetail detail = new ProcessorExecutionDetail(processor.getType());
+                if (request.source().verbosePipeline()) {
+                    detail.addInput(Arrays.asList(r.getHits().deepCopy().getHits()));
+                }
                 beforeResponseProcessor(processor);
                 final long start = relativeTimeSupplier.getAsLong();
                 processor.processResponseAsync(request, r, requestContext, ActionListener.wrap(rr -> {
                     long took = TimeUnit.NANOSECONDS.toMillis(relativeTimeSupplier.getAsLong() - start);
+                    if (request.source().verbosePipeline()) {
+                        detail.addOutput(Arrays.asList(rr.getHits().deepCopy().getHits()));
+                        detail.addTook(took);
+                        requestContext.addProcessorExecutionDetail(detail);
+                        rr.getInternalResponse().getProcessorResult().add(detail);
```

Comment:
Do you need to append to both the `requestContext` and to the `SearchResponse`? Adding it to the `requestContext` doesn't help at this point. (It looks like you're done reading from it by the time you get here, since you read from it on line 296, and this code runs bottom-to-top.)

As mentioned in my comment above, I would just wrap each of the processors, let them all write to the `PipelineProcessingContext`, then pass the details from that to the response just before calling `finalListener`.